File: working dir/Widgets/Image/Image.java
package Image;

import java.io.File;

import javafx.fxml.FXML;
import javafx.geometry.Pos;
import javafx.scene.layout.AnchorPane;

import Configuration.IsConfig;
import Layout.widget.Widget;
import Layout.widget.controller.FXMLController;
import Layout.widget.controller.io.IsInput;
import Layout.widget.feature.ImageDisplayFeature;
import gui.objects.image.Thumbnail;
import util.access.V;
import util.async.future.Fut;
import util.graphics.drag.DragUtil;

import static Layout.widget.Widget.Group.OTHER;
import static de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon.DETAILS;
import static javafx.geometry.Pos.CENTER;
import static util.async.Async.FX;
import static util.graphics.Util.setAnchor;
import static util.graphics.drag.DragUtil.installDrag;

/**
 * FXML Controller class
 *
 * @author Plutonium_
 */
@Widget.Info(
    author = "Martin Polakovic",
    programmer = "Martin Polakovic",
    name = "Image",
    description = "Shows a static image",
    howto = "Available actions:\n" +
            "    Drag & drop image : Set custom image",
    notes = "",
    version = "1.0",
    year = "2015",
    group = OTHER
)
public class Image extends FXMLController implements ImageDisplayFeature {

    @FXML AnchorPane root;
    private final Thumbnail thumb = new Thumbnail();

    @IsConfig(name = "Custom image", info = "Image file to display.")
    private File img = new File("");


    @Override
    public void init() {
        thumb.setBackgroundVisible(false);
        thumb.setBorderVisible(false);
        thumb.setDragEnabled(true);
        setAnchor(root,thumb.getPane(),0d);

        // drag&drop
        installDrag(
            root, DETAILS,"Display",
            e -> DragUtil.hasImage(e),
            e -> img!=null && img.equals(DragUtil.getImageNoUrl(e)),
            e -> {
                Fut<File> future = DragUtil.getImage(e);
                future.use(img -> showImage(img),FX)
                      .showProgress(!future.isDone(),getWidget().getWindow()::taskAdd)
                      .run();
            }
        );
    }

    @Override
    public void refresh() {
        thumb.loadImage(img);
    }

    @Override
    @IsInput("To display")
    public void showImage(File img_file) {
        thumb.loadImage(img_file);
        img = img_file==null ? new File("") : img_file;
    }

}

File: working dir/widgets/image/Image.kt
package image

import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon.DETAILS
import javafx.scene.input.KeyCode
import sp.it.pl.gui.objects.image.Thumbnail
import sp.it.pl.layout.widget.Widget
import sp.it.pl.layout.widget.Widget.Group.OTHER
import sp.it.pl.layout.widget.controller.SimpleController
import sp.it.pl.layout.widget.controller.io.IsInput
import sp.it.pl.layout.widget.feature.ImageDisplayFeature
import sp.it.pl.main.APP
import sp.it.pl.util.async.FX
import sp.it.pl.util.conf.IsConfig
import sp.it.pl.util.conf.cn
import sp.it.pl.util.conf.only
import sp.it.pl.util.graphics.drag.DragUtil
import sp.it.pl.util.graphics.drag.DragUtil.installDrag
import sp.it.pl.util.graphics.layFullArea
import sp.it.pl.util.validation.Constraint.FileActor.FILE
import java.io.File
import java.util.function.Consumer

@Widget.Info(
        author = "Martin Polakovic",
        name = "Image",
        description = "Shows a static image",
        howto = "Available actions:\n"+"    Drag & drop image : Set custom image",
        version = "1.0",
        year = "2015",
        group = OTHER
)
class Image(widget: Widget<*>): SimpleController(widget), ImageDisplayFeature {

    private val thumb = Thumbnail()

    @IsConfig(name = "Custom image", info = "Image file to display.")
    private var img by cn<File?>(null).only(FILE)

    override fun init() {
        thumb.isBackgroundVisible = false
        thumb.borderVisible = false
        thumb.isDragEnabled = true

        this layFullArea thumb.pane

        installDrag(
                this, DETAILS, "Display",
                { DragUtil.hasImage(it) },
                { e -> img!=null && img==DragUtil.getImageNoUrl(e) },
                { e -> DragUtil.getImage(e) ui { showImage(it) } }
        )
        setOnKeyPressed {
            if (it.code==KeyCode.ENTER) {
                APP.actions.openImageFullscreen(img)
                it.consume()
            }
        }
    }

    override fun refresh() = thumb.loadImage(img)

    @IsInput("To display")
    override fun showImage(imgFile: File) {
        img = imgFile
        thumb.loadImage(imgFile)
        requestFocus()
    }

}